splunk-adx-alert-addon/alert_actions_base.py

```diff
@@ -0,0 +1,249 @@
+from __future__ import print_function
+from builtins import str
+import csv
+import gzip
+import sys
+
+try:
+    from splunk.clilib.bundle_paths import make_splunkhome_path
+except ImportError:
+    from splunk.appserver.mrsparkle.lib.util import make_splunkhome_path
+
+# TODO: How does it depend on CIM module?
```

Comment:
Please find my comments as following
a) Some of these classes are self generated, as discussed I will remove them from the commit.
b) Yes Ram, I do agree. The only problem is invocation and execution since it is invoked from splunk main app. I checked with Sentinel team as well and they had similar concerns. We can have separate main method for execution of these methods but the problem will be maintaining them and keeping them insync with the actual files.
c) you can ignore them, since these are self generated files